exercises/luhn/src/test/java/LuhnValidatorTest.java

```diff
@@ -73,26 +73,38 @@ public void testThatStringContainingPunctuationIsInvalid() {
     */
     @Ignore("Remove to run test")
     @Test
-    public void testThatStringContainingSymbolsIsInvalid() {
-        assertFalse(luhnValidator.isValid("34&"));
+    public void testStringContainingSymbolsInvalidJavaTrackSpecific() {
```

Comment:
I think this should be moved so that it's the last test. Otherwise I think fixing this test might also fix later tests which we'd rather avoid. What do you think @c-thornton @sjwarner-bp? 🙂

Reply:
I am not sure if it should be moved to the bottom, but I see where you are coming from with the ordering. I kept all of the tests in the order that they appear in the canonical data and I slid this one in where there was a similar test in there.

Putting the tests in an order that will check in an increasing depth may provide better incremental improvements to the users program. I could put them in an order like the below if you think that would be better.
* testSingleDigitStringInvalid
* testSingleZeroIsInvalid
* testSingleSpaceWithZeroInvalid
* testMoreThanSingleZeroValid
* testDigitNineConvertedToOutputNine
* testSimpleValidSINReversedRemainsValid
* testSimpleValidSINReversedBecomesInvalid
* testValidCanadianSINValid
* testInvalidCanadianSINInvalid
* testInvalidCreditCardInvalid
* testStringsContainingNonDigitInvalid
* testStringContainingPunctuationInvalid
* testStringContainingSymbolsInvalid
* testStringsWithNonDigitsInvalid
* testStringContainingSymbolsInvalidJavaTrackSpecific

Reply:
In general we prefer to stick to the order of the test in the canonical data (like you've done in this PR) so that it's consistent across the different tracks 🙂 But if you think of any improvements to the order feel free to open an issue or a PR in [problem specifications](https://github.com/exercism/problem-specifications) to change the order in the canonical data 🙂 

However, since this specific test is just for the Java track we can decide where it should go 🙂 I think it would make sense for it to go at the end since it's testing for quite a specific bug and can therefore come after the tests for the core functionality. What do you think @c-thornton? 🙂

Reply:
That makes sense to me as to why it should be at the end instead of in the middle where it currently is. I will move that down tonight while I am taking care of the indentation as well and then it should be ready to go. Thanks for the breakdown @FridaTveit I appreciate it. :smile: